dagfactory/dagbuilder.py

```diff
@@ -549,52 +549,102 @@ def make_task_groups(task_groups: Dict[str, Any], dag: DAG) -> Dict[str, "TaskGr
         task_groups_dict: Dict[str, "TaskGroup"] = {}
         if version.parse(AIRFLOW_VERSION) >= version.parse("2.0.0"):
             for task_group_name, task_group_conf in task_groups.items():
-                task_group_conf["group_id"] = task_group_name
-                task_group_conf["dag"] = dag
+                circularity_check_queue = []
+                DagBuilder.make_nested_task_groups(
+                    task_group_name, task_group_conf, task_groups_dict, task_groups, circularity_check_queue, dag
+                )
```

Comment:
```suggestion
                DagBuilder.make_nested_task_groups(
                    task_group_name, task_group_conf, task_groups_dict, task_groups, None, dag
                )
```
maybe good to pass circularity_check_queue as None, and in the method make_nested_task_groups, we could do 
```
if circularity_check_queue is None:
    circularity_check_queue = []
```